Sources/Workspace/Workspace.swift

```diff
@@ -942,16 +947,20 @@ public class Workspace {
     /// Validates that all the edited dependencies are still present in the file system.
     /// If some edited dependency is removed from the file system, mark it as unedited and
     /// fallback on the original checkout.
-    private func validateEditedPackages() throws {
+    private func fixManagedDependencies() throws {
         let managedDependencies = try self.managedDependencies.load()
         for dependency in managedDependencies.values {
-            switch dependency.state {
-            case .checkout: continue
-            case .edited:
-                let dependencyPath = path(for: dependency)
-                
-                // If some edited dependency has been removed, mark it as unedited.
-                if !fileSystem.exists(dependencyPath) {
+            let dependencyPath = path(for: dependency)
+            if !fileSystem.isDirectory(dependencyPath) {
+                switch dependency.state {
+                case .checkout(let checkoutState):
+                    // If some checkout dependency has been removed, clone it again.
+                    _ = try clone(repository: dependency.repository, at: checkoutState)
+                    // FIXME: Use diagnosics engine when we have that.
+                    let dependencyName = dependency.repository.url.components(separatedBy: "/").last!
```

Comment:
Not sure how to write the test now. The warning message is now random due to repo's url being random (fixture).